Homework-2-In-Memory-CRUD-REST-Service/src/routers/userRouter.js

```diff
@@ -0,0 +1,35 @@
+import express from 'express';
+import Joi from 'joi';
+import { createValidator } from 'express-joi-validation';
+import {
+    getAllUsers,
+    getUserById,
+    createUser,
+    updateUser,
+    deleteUser
+} from '../controllers/userController.js';
+
+const schema = Joi.object({
+    login: Joi.string()
+        .alphanum()
+        .trim()
+        .required(),
+    password: Joi.string()
+        .pattern(new RegExp('^(?=.*[A-Za-z])(?=.*\\d)[A-Za-z\\d]{8,}$'))
+        .required(),
+    age: Joi.number()
+        .integer()
+        .min(4)
+        .max(130)
+        .required()
+});
+
+const validator = createValidator({ passError: true });
+
+export const usersRouter = express.Router();
+
+usersRouter.get('/', getAllUsers);
+usersRouter.get('/:id', getUserById);
+usersRouter.post('/', validator.body(schema), createUser);
```

Comment:
I don't know at what layer there is a problem but when I send empty object in body it just shows me 500. Instead of message with missing fields, etc.

Even if I provide all fields, I still see 500

Reply:
Yeah, now I see the same and it's weird, I use `{passError: true}` while initializing validator and it should forward error to my error handler middleware, but it doesn't. @boccob Any thoughts why it could happen?